src/panels/ConfigPanel/ConfigPanel.tsx

```diff
@@ -277,6 +350,67 @@ export default function ConfigPanel(): JSX.Element {
           </Portal>
         );
       }
+      if (confirmType === ConfigConfirmType.GRANT) {
+        // Render the grant confirmation modal.
+        const requiredGrants = getRequiredGrants(appName, config, secrets);
+        return (
+          <Portal>
+            <ConfirmationModal
+              // Prevent clicks inside this panel from closing the parent panel.
+              // This is handled in `checkCanClose`.
+              className="prevent-panel-close"
+              title={Label.GRANT_CONFIRM}
+              cancelButtonLabel={Label.GRANT_CANCEL_BUTTON}
+              confirmButtonLabel={Label.GRANT_CONFIRM_BUTTON}
+              confirmButtonAppearance="positive"
+              onConfirm={() => {
+                setConfirmType(null);
+                // Clear the form errors if there were any from a previous submit.
+                setInlineError(InlineErrors.FORM, null);
+                if (!appName || !requiredGrants) {
+                  // It is not possible to get to this point if these
+                  // variables aren't set.
+                  return;
```

Comment:
This part was not tested, but I am not sure if it is worth adding a test here anyway. What do you think?

Reply:
There's no way to actually get to this code (there's no way to display this confirmation if there isn't an app name or required grants), it just needs this return to satisfy typescript.

Reply:
Oh, sorry, didn't read the comment that was literally there. :) Thanks for the explanation!